src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/AuditUtility.cs

```diff
@@ -281,5 +290,45 @@ private int ParseAuditLevel()
             _logger.Log(message);
             return 1;
         }
+
+        private enum NuGetAuditMode { Unknown, Direct, All }
+
+        private NuGetAuditMode ParseAuditMode()
+        {
+            string? auditMode = _restoreAuditProperties.AuditMode?.Trim();
+
+            if (auditMode == null)
+            {
+                return NuGetAuditMode.Unknown;
+            }
+            else if (string.Equals("direct", auditMode, StringComparison.OrdinalIgnoreCase))
+            {
+                return NuGetAuditMode.Direct;
+            }
+            else if (string.Equals("all", auditMode, StringComparison.OrdinalIgnoreCase))
+            {
+                return NuGetAuditMode.All;
+            }
+
+            string messageText = string.Format(Strings.Error_InvalidNuGetAuditModeValue, auditMode, "direct, all");
+            RestoreLogMessage message = RestoreLogMessage.CreateError(NuGetLogCode.NU1014, messageText);
+            message.ProjectPath = _projectFullPath;
+            _logger.Log(message);
+            return NuGetAuditMode.Unknown;
+        }
+
+        IEnumerable<ResolvedDependencyKey> GetDirectDependencies(RestoreTargetGraph graph)
+        {
+            Debug.Assert(graph.Graphs.Count() == 1);
```

Comment:
Can't there be 0 here? I see some calls (eg, in `RestoreCommand`) to create `RestoreTargetGraph` which passes an empty enumerable for `graphs`.

Reply:
Good find!  Looks like graphs can be empty when there are certain error conditions, which I didn't test (other error conditions still generate a graph). Assuming I correctly commit and pushed this time, this code should be resilient to those error conditions now.